### PR TITLE
v1.1 [WIP]

### PR DESCRIPTION
**Pull request by laullon**

### CODE REVIEW

File: nozzle/api_client.go [hunk 1]
```diff
@@ -41,17 +42,23 @@ func newAppInfo(app cfclient.App) *AppInfo {
 
 // NewAPIClient crate a new ApiClient
 func NewAPIClient(conf *NozzleConfig) (*APIClient, error) {
-	apiURL := conf.APIURL
+	apiURL := strings.Trim(conf.APIURL, " ")
```
Comment (vikramraman) [inline nozzle/api_client.go:45]: Fine for this PR. But going forward, whatever is needed in master, let's create a PR against master first. We can merge those changes into this branch from master. Also if anything from here is needed in master, let's create one this one time after this PR is merged.

File: nozzle/event_handler.go [hunk 2]
```diff
@@ -81,26 +90,48 @@ func CreateEventHandler(conf *WavefrontConfig) EventHandler {
...
+	}
+
+	logger.Printf("internalTags: %v", internalTags)
+
+	numMetricsSent := newCounter("total-metrics-sent", internalTags)
+	metricsSendFailure := newCounter("metrics-send-failure", internalTags)
+	numValueMetricReceived := newCounter("value-metric-received", internalTags)
+	numCounterEventReceived := newCounter("counter-event-received", internalTags)
+	numContainerMetricReceived := newCounter("container-metric-received", internalTags)
+	handleErrorMetric := newCounter("handleError", internalTags)
```
Comment (vikramraman) [inline nozzle/event_handler.go]: rename to `firehose-error` or `connection-error`.

File: nozzle/nozzle.go [hunk 3]
```diff
@@ -86,13 +86,9 @@ func (s *forwardingNozzle) handleEvent(envelope *events.Envelope) {
 	case events.Envelope_ContainerMetric:
 		appGuIG := envelope.GetContainerMetric().GetApplicationId()
 		appInfo, err := s.fetcher.GetApp(appGuIG)
-		if err != nil {
+		if err != nil && debug {
```
Comment (vikramraman) [inline nozzle/nozzle.go:89]: Let's emit a metric `appinfo-fetch-error` for this case.

File: nozzle/api_client.go (hunk 1 above)
Comment (laullon) [reply nozzle/api_client.go:45]: they are in other branch too...

https://github.com/wavefrontHQ/cloud-foundry-nozzle-go/commits/v1.0.1

**Pull request merged**